cloudinit/config/cc_ubuntu_advantage.py

```diff
@@ -199,29 +200,40 @@ def configure_ua(token=None, enable=None, config=None):
                 ", ".join('"{}"'.format(param) for param, _ in enable_errors)
             )
         )
-    attach_cmd = ["ua", "attach", token]
+
+    if enable:
+        attach_cmd = ["ua", "attach", "--no-auto-enable", token]
+    else:
+        attach_cmd = ["ua", "attach", token]
     LOG.debug("Attaching to Ubuntu Advantage. %s", " ".join(attach_cmd))
     try:
-        subp.subp(attach_cmd)
+        # Allow `ua attach` to fail in already attached machines
+        subp.subp(attach_cmd, rcs={0, 2})
     except subp.ProcessExecutionError as e:
         msg = "Failure attaching Ubuntu Advantage:\n{error}".format(
             error=str(e)
         )
         util.logexc(LOG, msg)
         raise RuntimeError(msg) from e
+
     enable_errors = []
     for service in enable:
         try:
             cmd = ["ua", "enable", "--assume-yes", service]
             subp.subp(cmd, capture=True)
         except subp.ProcessExecutionError as e:
-            enable_errors.append((service, e))
+            if re.search("is already enabled.", str(e)):
```

Comment:
If using `--format json` above we can source json output of e.stdout to determine the failure type instead of patial matching the English string.

```suggestion
            error_codes = [e.get("message_code") for e in json.loads(e.stdout)["errors"]]
            if 'service-already-enabled' in error_codes:
```

Reply:
@chifac08 thanks so much for this contribution. My minor PR change request can be carried forward to a separate PR, I'll dismiss my comment here and we can land your work. Again thanks for patience and effort here to improve cloud-init and UA integration.